### DISCUSSION TITLE
P(s) curve and its derivative on individual chromosomes

### DISCUSSION THREAD

**Discussion by LAlvarezGlez** [open2c/cooltools#315] [Q&A] [answered]
Hi!

is it possible to calculate the Ps curve and its derivative per chromosome? In the notebook is written how to do it by regions, but I don't understand how to do it by chromosomes.

Many thanks!

Lucía

**Comment by golobor** [comment, marked as answer]
sure!

You should be able to re-use the code for derivative plotting from our [contacts-vs-distance notebook](https://cooltools.readthedocs.io/en/latest/notebooks/contacts_vs_distance.html) (currently, cells 13-15) with two small modifications:

1. Filter the cvd dataframe to select only the chromosomal region of interest:

```
cvd_merged = cvd_merged[cvd_merged['name'] == 'chr2_p']
```

2. Instead of 'balanced.avg.smoothed.agg' (which contains genome-wide aggregated contact frequencies), use 'balanced.avg.smoothed'  (non-aggregated contact frequencies).

```
der = np.gradient(np.log(cvd_merged['balanced.avg.smoothed']),
                  np.log(cvd_merged['s_bp']))
f, axs = plt.subplots(
    figsize=(6.5,13),
    nrows=2,
    gridspec_kw={'height_ratios':[6,2]},
    sharex=True)
ax = axs[0]
ax.loglog(
    cvd_merged['s_bp'],
    cvd_merged['balanced.avg.smoothed'],
    '-',
    markersize=5,
)

ax.set(
    ylabel='IC contact frequency',
    xlim=(1e3,1e8)
)
ax.set_aspect(1.0)
ax.grid(lw=0.5)


ax = axs[1]
ax.semilogx(
    cvd_merged['s_bp'],
    der,
    alpha=0.5
)

ax.set(
    xlabel='separation, bp',
    ylabel='slope')

ax.grid(lw=0.5)

```

Please, let us know if this explanation was clear and if it works!
